scripts/tutorial-uploader/lib/local-tutorial-data.ts

```diff
@@ -0,0 +1,73 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import yaml from "js-yaml";
+import { readFile } from "fs/promises";
+import { dirname, join } from "path";
+import { isString, isNumber, isBoolean } from "lodash";
+
+/* Information specified in the YAML file */
+export interface LocalTutorialData {
+  title: string;
+  short_description: string;
+  slug: string;
+  status: string;
+  local_path: string;
+  category: string;
+  topics: string[];
+  reading_time: number;
+  catalog_featured: boolean;
+}
+
+function relativiseLocalPath(
+  tutorial: LocalTutorialData,
+  path: string,
+): LocalTutorialData {
+  tutorial.local_path = join(dirname(path), tutorial.local_path);
+  return tutorial;
+}
+
+export async function readTutorialData(
+  path: string,
+): Promise<LocalTutorialData[]> {
+  const raw = await readFile(path, "utf8");
+  const parsed = yaml.load(raw) as unknown[];
+  parsed.forEach((i) => assertIsLocalTutorialData(i));
+  return parsed.map((i) => relativiseLocalPath(i as LocalTutorialData, path));
```

Comment:
Is the `as` still required, thanks to the type guard from assertIsLocalTutorialData?

```suggestion
  return parsed.map((i) => relativiseLocalPath(i, path));
```

Reply:
I tried that but unfortunately the type checker isn't that clever. It did work when I made a function that asserted `parsed` was of type `LocalTutorialData[]`, but I thought this was just as clear and a little simpler.